### ISSUE TITLE
Stress Test The Current Game Engine 

### ISSUE THREAD

**Issue by Ridleh** [Duelers/Duelers#562]
While testing the new abyssian cards, I ran into a lot of concurrentModificationExceptions or NPEs, with the chances of receiving these errors increasing as the number of cards on the board increasing.

Additionally, long chain effects (e,g, minion A attacks minion B. Minion B deals 3 damage around it, killing minion C. minion C has a dying wish effect that kills a random minion near it, killing minion A. minion A has a dying wish affect that heals its general), although rare, seem to not always work consistently

We should come up with more test edge cases and see what our current game engine is capable of and fix/modify any areas thats lacking. 

orwecanpushupqazz'senginerevampuphigherontheprioritylist

**Comment by fluffy-hamster** [comment]
@Ridleh  
Is this because of end of turn mechanics?

If its hard, it maybe be better to just ditch these effects and wait for Quazz's system.

Remember these cards are only supposed to be temporary, so its a mistake for us to invest huge amounts of time in fixing the system.

**Comment by Ridleh** [comment]
@fluffy-hamster no, these bugs occurred while testing out design team's new abyss cards. 

Most notable bug is when one cacophynos kills another, the game silently breaks (the terminal prints non stop NPEs)
